.github/workflows/self-hosted-gpu-test.yml

```diff
@@ -0,0 +1,126 @@
+name: self-hosted-gpu-test
+on:
+  push:
+    branches:
+      - master
+      - feat/add_aws_testing 
+  workflow_dispatch:
+  schedule:
+    # weekly tests
+    - cron: "0 0 * * SUN"
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+jobs:
+  start-runner:
+    name: Start self-hosted EC2 runner
+    runs-on: ubuntu-latest
+    outputs:
+      label: ${{ steps.start-ec2-runner.outputs.label }}
+      ec2-instance-id: ${{ steps.start-ec2-runner.outputs.ec2-instance-id }}
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+      - name: Try to start EC2 runner
+        id: start-ec2-runner
+        uses: machulav/ec2-github-runner@main
+        with:
+          mode: start
+          github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          ec2-image-id: ami-04d16a12bbc76ff0b
+          ec2-instance-type: g4dn.xlarge
+          subnet-id: subnet-0dee8543e12afe0cd # us-east-1a
+          security-group-id: sg-0f9809618550edb98
+          # iam-role-name: self-hosted-runner # optional, requires additional permissions
+          aws-resource-tags: > # optional, requires additional permissions
+            [
+              {"Key": "Name", "Value": "ec2-github-runner"},
+              {"Key": "GitHubRepository", "Value": "${{ github.repository }}"}
+            ]
+
+  do-the-job:
+    name: Do the job on the runner
+    needs: start-runner # required to start the main job when the runner is ready
+    runs-on: ${{ needs.start-runner.outputs.label }} # run the job on the newly created runner
+    timeout-minutes: 1200 # 20 hrs
+    steps:
+
+    
+    - name: Check out
+      uses: actions/checkout@v3
+
+    - name: Install Miniconda
+      uses: conda-incubator/setup-miniconda@v2
+      env:
+        HOME: /home/ec2-user
+      with:
+        activate-environment: ""
+        auto-activate-base: true
+        miniforge-variant: Mambaforge
+          
+    - name: Prepare dependencies (with CUDA)
+      run: |
+        sed -i -e "/cudatoolkit/c\  - cudatoolkit 11.7.*" \
+               -e "/gxx_linux-64/c\  - gxx_linux-64 10.3.*" \
+               -e "/torchani/c\  - torchani 2.2*" \
+               -e "/nvcc_linux-64/c\  - nvcc_linux-64 11.7.*" \
+               -e "/python/c\  - python 3.10.*" \
+               -e "/pytorch-gpu/c\  - pytorch-gpu 2.0.*" \
+               environment.yml
```

Comment:
This should somehow use the same values as the current ci.yml, even if it is only for the latest versions.
Do you guys think it is worth joining this yml and the current ci.yml?

Reply:
Is the environment yaml that is in the root of the repo an example of the "latest" verions of pins? I was going to refactor this a bit like I did here: https://github.com/openmm/openmm-torch/pull/107/files#diff-bd64c4de8527b7011b64ffef8a82f8ab32dae8c0ac01c59a71d4f3b28f05cfddR47-R54 so that it will be easy to modify these and update to the latest versions, much like how the github CI is setup.

Reply:
I was thinking about this:
https://github.com/openmm/NNPOps/blob/b63fc70e9b671988f67aec1fbc62bbd01f830b25/.github/workflows/ci.yml#L22-L61
Which transforms the env yaml like this:
https://github.com/openmm/NNPOps/blob/b63fc70e9b671988f67aec1fbc62bbd01f830b25/.github/workflows/ci.yml#L83-L90
So basically what you mentioned.